### ISSUE TITLE
Erratic Behavior with $.trim(NaN) in Responsive View with Chrome

### ISSUE THREAD

**Issue by bastiantowers** [madrobby/zepto#1195]
Using the Viewport Chrome Emulator and sets the "user-agent" as "Responsive", if you pass NaN as a parameter of $.trim(), throws "Uncaught TypeError: str.trim is not a function".

Is a Zepto issue? or it is a Chrome bug?

Thanks!!!

**Comment by noBlubb** [comment]
http://stackoverflow.com/questions/115548/why-is-isnannull-false-in-js

Quick fix would be something like this:
```function trim(str) {
  if (str) {
    return String.prototype.trim.call(str)
  }
  return "";
}```

**Comment by bastiantowers** [comment]
@noBlubb Shouldn't be better if the function returns `false`, instead `""`? You would know if something went wrong when you execute the `trim` function and maybe make some validations wrapped in an if condition...

**Comment by noBlubb** [comment]
No, that's not how trim functions (in other languages) usually work. C#
offers custom String.IsEmpty functions, but that's an overly complicated
solution for this problem. You can always check `if (fuubar)` and check if
it turns out to be true(-ish).

If trim could return a boolean value aswell, you end up with broken APIs as
in PHP, where some built-ins return booleans and strings.

Bastian <notifications@github.com> schrieb am Do., 23. Juni 2016, 20:57:

**Comment by bastiantowers** [comment]
@noBlubb Seems reasonable! Thanks for the explanation! Maybe this fix should be included directly in Zepto library, to prevent this to happen and to avoid to overwrite the trim function externally. I will suggest this with a PR and I will share with you. Thanks for your time!

**Comment by mislav** [comment]
This was fixed in f3706d63597c175d6728c2caf3be1a765bd69524 and released as part of v1.1.0.
